packages/pigeon/example/app/android/app/src/main/java/io/flutter/plugins/Messages.java

```diff
@@ -268,11 +268,19 @@ public interface ExampleHostApi {
     }
     /** Sets up an instance of `ExampleHostApi` to handle messages through the `binaryMessenger`. */
     static void setUp(@NonNull BinaryMessenger binaryMessenger, @Nullable ExampleHostApi api) {
+      setUp(binaryMessenger, "", api);
+    }
+
+    static void setUp(
+        @NonNull BinaryMessenger binaryMessenger,
+        @NonNull String messageChannelSuffix,
+        @Nullable ExampleHostApi api) {
       {
         BasicMessageChannel<Object> channel =
             new BasicMessageChannel<>(
                 binaryMessenger,
-                "dev.flutter.pigeon.pigeon_example_package.ExampleHostApi.getHostLanguage",
+                "dev.flutter.pigeon.pigeon_example_package.ExampleHostApi.getHostLanguage"
+                    + messageChannelSuffix,
```

Comment:
I was thinking about this again after offline discussion, and I think we actually need to add the separator ourselves even though it'll make the code more complicated. Otherwise the following could happen:
- API declares a method called `foo`, and another called `foobar`.
- A non-suffixed version of the API is instantiated.
- A suffixed version is also instantiated, with the suffix `bar`.

Now Pigeon has tried to register two handlers for the same name: `dev.flutter.pigeon.package_name.ApiName.foobar`

This is an edge case, but with dynamic suffixes based on something other than a counter it's not impossible to imagine happening, and it would be *really* confusing for clients if it ever did. And we can prevent that edge case by just adding the separator any time there's a non-empty suffix.

Reply:
Makes perfect sense.